CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# WIP: Internal contributing guide
+
+## Weave Gitops :heart:s your contributions
+
+Thank you for taking the time to contribute to Weave Gitops!
+
+This guide **is a work in progress** but aims to cover all aspects of how to
+interact with the project and how to get involved in development as smoothly as possible.
+
+If we have missed anything you think should be included, or if anything is not
+clear, we also accept contributions to this contribution doc :smile:.
+
+
+## Building the binary
+To build the `gitops` binary locally you can run `make bin`. This will create a `gitops`
+binary in the `bin/` directory.
+
+
+## Testing
+We have 3 layers of testing:
+- Unit testing
+- Integration testing
+- Acceptance testing
+
+We aim to follow the [testing pyramids best practises](https://martinfowler.com/articles/practical-test-pyramid.html)
+with the focus on a large number of units tests, fewer integration tests and even fewer acceptance tests.
+
+
+### Running unit tests
+You can run `make unit-tests` from the top of the repository to run all of the unit tests.
+As part of running the unit tests we will install any required dependencies for building
+the code. See [dependencies](tools/dependencies.toml) for more information on whats installed.
+
+
+### Running integration tests
+The integration tests can be run using `make integration-tests`. They require some
+extra environment variables to run:
+
+- `GITHUB_ORG`- The name of the org that test repositories can be created in.
+if you're a weaveworks employee you can use `weaveworks-gitops-test`
+- `GITHUB_TOKEN`- Used to create, push and delete repositories. Please ensure
+the token has sufficient permissions, as deleting repositories is not part
+of the standard set of permissions a token is given.
+
+If you wish to run the full suite, you will also need the equivalent for gitlab:
+- `GITLAB_TOKEN`
+- `GITLAB_ORG`
```

Comment:
I assume the equivalent also means that there are weavework test repos that one can use as with Github?

Reply:
no idea :see_no_evil: never used/touched gitlab